jsdom-nodejs-env/src/test/scala/org/scalajs/jsenv/jsdomnodejs/JSDOMNodeJSEnvTest.scala

```diff
@@ -21,4 +30,105 @@ class JSDOMNodeJSEnvTest {
         .expectOut("http://localhost/foo\n")
     }
   }
+
+  @Test
+  def reactUnhandledExceptionHack_issue42: Unit = {
+    val code =
+      """
+      |const rootElement = document.createElement("div");
+      |document.body.appendChild(rootElement);
+      |
+      |class ThrowingComponent extends React.Component {
+      |  render() {
+      |    throw new Error("boom");
+      |  }
+      |}
+      |
+      |class ErrorBoundary extends React.Component {
+      |  constructor(props) {
+      |    super(props);
+      |    this.state = { hasError: false };
+      |  }
+      |
+      |  componentDidCatch(error, info) {
+      |    this.setState({error: error.message, hasError: true});
+      |  }
+      |
+      |  render() {
+      |    if (this.state.hasError) {
+      |      console.log("render-error");
+      |      return React.createElement("p", null,
+      |        `Caught error: ${this.state.error}`);
+      |    } else {
+      |      return this.props.children;
+      |    }
+      |  }
+      |}
+      |
+      |class MyMainComponent extends React.Component {
+      |  render() {
+      |    console.log("two");
+      |    return React.createElement(ErrorBoundary, null,
+      |      React.createElement(ThrowingComponent)
+      |    );
+      |  }
+      |}
+      |
+      |console.log("begin");
+      |
+      |const mounted = ReactDOM.render(
+      |  React.createElement(ErrorBoundary, null,
+      |    React.createElement(ThrowingComponent, null)
+      |  ),
+      |  rootElement
+      |);
+      |
+      |console.log(document.querySelector("p").textContent);
+      |
+      |console.log("end");
+      """.stripMargin
+
+    kit.withRun(ReactJSFiles :+ codeToInput(code)) {
+      _.expectOut("begin\nrender-error\nCaught error: boom\nend\n")
+        .succeeds()
+    }
+  }
+}
+
+object JSDOMNodeJSEnvTest {
+  private lazy val ReactJSFiles: List[Input] = {
+    val fs = Jimfs.newFileSystem()
```

Comment:
Only create one `fs` for the test?

Reply:
We can but then we'd have to recreate the `ReactJSFiles` for other tests that want it (if any). We can't share filesystems between 2 tests, as that would make them non-independent.

Reply:
Fair enough :)